include/circt/Dialect/HW/HWMiscOps.td

```diff
@@ -53,6 +54,52 @@ def ConstantOp
   let hasVerifier = 1;
 }
 
+def WireOp : HWOp<"wire", [
+    SameOperandsAndResultType,
+    DeclareOpInterfaceMethods<OpAsmOpInterface, ["getAsmResultNames"]>]> {
+  let summary = "Assign a name or symbol to an SSA edge";
+  let description = [{
+    An `hw.wire` is used to assign a human-readable name or a symbol for remote
+    references to an SSA edge. It takes a single operand and returns that
+    its value unchanged as a result. The operation guarantees the following:
+
+    - The value of its operand remains observable under the given human-readable
+      name in any final output, or under the given symbol within the IR.
```

Comment:
Maybe add something like "if specified" to be clear that name and inner_sym are optional, and this behavior is only present if one or the other is specified? Also, "remains observable" may be too strong... it looks like the canonicalizer will try to propagate the name attribute as a sv.namehint, but this is only a hint, and doesn't guarantee that the name will be observed.

Reply:
Yeah great point! I'll reword this to match exactly how the canonicalizer works right now. At some point we might want to specifically indicate on the op if a name is droppable or *must* be preserved. The way it is right now preserves how `sv.wire` behaves, which seems like a good starting point.